DESIGN.md

```diff
@@ -73,77 +157,225 @@ Flow control support:
+Relation to Reactive Streams:
 
-Flow control support:
+- It can not implement Reactive Streams `Publisher` unless it is created with a default consumer-driven flow control strategy.
+- It can not implement `Processor` since a `Processor` must compose `request(n)` which can not be done with multicasting or push.
+
+Here is an approach to converting from a `Subject` to Reactive Streams types by adding a default flow control strategy:
+
+```java
+Subject s = PublishSubject.create();
+// convert to Publisher with backpressure strategy
+Publisher p = s.toPublisher(onBackpressureStrategy);
+
+// now the request(n) semantics are handled by default
+p.subscribe(subscriber1); 
+p.subscribe(subscriber2);
+```
+
+In this example, `subscriber1` and `subscriber2` can consume at different rates, `request(n)` will propagate to the provided  `onBackpressureStrategy`, not the original `Subject` which can't propagate `request(n)` upstream.
 
-- buffering, sampling, throttling, windowing, dropping, etc
-- temporal and count-based strategies
-- It does not support pull-based consumer-driven flow control.
 
 ##### Disposable
 
 A type representing work that can be cancelled or disposed.
 
-### Behavior
+Examples:
 
-##### Creation
+- An `Observable.subscribe` passes a `Disposable` to the `Observable.onSubscribe` to allow the `Observer` to dispose of the subscription. 
+- A `Scheduler` returns a `Disposable` that you use for disposing of the `Scheduler`.
 
-Creation of a stream falls into the following use cases, all of which should be catered to in API design.
+`Subscription` is a similar type used for lifecycle management on the `Flowable` type with interactive flow control.
 
-- async, hot, push (ie. system or user events)
-- async, cold, push (ie. events resulting from remote system via network connection)
-- sync, cold, pull (ie. iterable, file, range)
-- async, cold, pull (ie. RPC/REST network call, cross-thread queue draining)
- 
-Unknown:
+##### Operator
 
-- hot, pull (what is an example of this?)
+An operator follows a specific lifecycle (union of the producer/consumer contract).
 
-Flow control support:
+- It must propagate the `subscribe` event upstream (to the producer).
+- It must obey the RxJava contract (serialize all events, `onError`/`onComplete` are terminal).
+- If it has resources to cleanup it is responsible for watching `onError`, `onComplete`, and `cancel/dispose`, and doing the necessary cleanup.
+- It must propagate the `cancel/dispose` upstream.
+
+In the addition of the previous rules, an operator for `Flowable`:
+
+- It must propagate/negotiate the `request(n)` event.
+
+
+### Creation
+
+Creation of the various types should be exposed through factory methods that provide safe construction.
+
+```java
+Flowable.create(SyncGenerator generator)
+
+Flowable.create(AsyncGenerator generator)
+
+Observable<T>.create(OnSubscribe<Observer<T>> onSubscribe)
```

Comment:
Yes in this model it seems like `OnSubscribe` is an `Action0` (since `Consumer` is a Java1.8 only functional interface). Now that you've had a chance to look at the `Consumable` proposal a little closer do you still share this opinion? I don't think that I have a strong opinion here.